### PR TITLE
added the user agent to req header

### PR DESCRIPTION
**Pull request by ejahnGithub**
Signed-off-by: eugene <ejahngithub@github.com>

#### Summary
Add user agent in request header, so the server side will know the request is from sigstroe-js. 

Issue: https://github.com/sigstore/sigstore-js/issues/32

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
   "extends": "@tsconfig/node12/tsconfig.json",
   "compilerOptions": {
+    "resolveJsonModule": true,
     "noImplicitAny": true,
     "outDir": "./dist"
   },
   "include": [
-    "src/**/*"
+    "src/**/*",
+    "package.json"
```
Comment (bdehamer) [inline tsconfig.json:10]: I'm not sure this is necessary, I believe that the "package.json" will always be included by default.

**Review by bdehamer — CHANGES_REQUESTED** [review]
This is a great first pass. I've got a couple of clean-up items in the comments below. Also, let's be sure to add a unit test for the `getUserAgent` function. We should also be able to update some of the existing tests for the Fulctio/Rekor clients to ensure that the new user agent header is being included.

File: tsconfig.json (hunk 1 above)
Comment (ejahnGithub) [reply tsconfig.json]: solved

File: src/util.ts [hunk 2]
```diff
@@ -55,3 +58,39 @@ export const promiseAny = async <T>(
...
+    case 'openbsd':
+      return 'OpenBSD platform';
+    case 'sunos':
+      return 'SunOS platform';
+    case 'win32':
+      return 'windows platform';
+    default:
+      return 'unknown platform';
+  }
+};
```
[the remark this answers is not in the document]
Comment (ejahnGithub) [reply src/util.ts]: solved

File: bin/sigstore.js [hunk 3]
```diff
@@ -14,4 +14,4 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-require('../dist/cli').processArgv();
+require('../dist/src/cli').processArgv();
```
[the remark this answers is not in the document]
Comment (ejahnGithub) [reply bin/sigstore.js:17]: umm. i got this error message when i am building 
```
node:internal/modules/cjs/loader:959
  throw err;
  ^

Error: Cannot find module '../dist/cli'
```

File: src/util.ts [hunk 4]
```diff
@@ -55,3 +58,10 @@ export const promiseAny = async <T>(
     (value) => Promise.resolve<T>(value)
   );
 };
+
+// Format User-Agent: <product> / <product-version> (<platform>)
+// source: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent
+export const getUserAgent = (): string => {
+  const platformName = os.platform();
+  return `sigstore-js/${packageJson.version} (${platformName})`;
```
Comment (feelepxyz) [inline src/util.ts:66]: Thoughts on also including the node version and architecture alongside the platform? Not sure it needs to be blocking, could be a follow up maybe?

Comment (feelepxyz) [reply src/util.ts:66]: not sure if we need to account for non-node envs like deno

Comment (ejahnGithub) [reply src/util.ts]: sounds good! i can add it as my next pr :)

**Review by bdehamer — APPROVED** [review]
Nicely done!

**Pull request merged**